src/ActualBlockEmitter.js

```diff
@@ -0,0 +1,26 @@
+const EventEmitter = require('events');
+const web3 = require('./utils/createAndUnlockWeb3');
+const logger = require('../src/config/winston');
+
+
+class ActualBlockEmitter extends EventEmitter {
+  constructor() {
+    super();
+    this.lastBlock = 0;
+  }
+
+  listen() {
+    setInterval(this.checkForNewBlock.bind(this), 5000);
+  }
+
+  async checkForNewBlock() {
+    const blockNumber = await web3.eth.getBlockNumber();
+    if (blockNumber > this.lastBlock) {
```

Comment:
When would this condition be false? In case of chain reorganization? Right now we ignore it?

Reply:
It's http long polling. Because of that there is fixed interval (5 seconds) between checks. In order to not emit the same events twice, we check if during last 5 seconds there has appeared new block.